lib/spack/spack/fetch_strategy.py

```diff
@@ -322,7 +322,22 @@ def fetch(self):
 
     def _existing_url(self, url):
         tty.debug('Checking existence of {0}'.format(url))
+        # Telling urllib to check if url is accessible
         curl = self.curl
+
+        if spack.config.get('config:use_curl') and curl is not None:
+            # Telling curl to fetch the first byte (-r 0-0) is supposed to be
+            # portable.
+            curl_args = ['--stderr', '-', '-s', '-f', '-r', '0-0', url]
+            _ = curl(*curl_args, fail_on_error=False, output=os.devnull)
+            return curl.returncode == 0
+        else:
+            try:
+                url, headers, response = web_util.read_from_url(url)
+            except web_util.SpackWebError:
+                msg = "Urllib fetch failed to verify url {0}".format(url)
+                raise FailedDownloadError(url, msg)
+            return (response.getcode() is None or response.getcode() == 200)
         # Telling curl to fetch the first byte (-r 0-0) is supposed to be
```

Comment:
This line to the end of the function has been copied to the if use_curl section above, should be removed here